### PR TITLE
[XamlC] Test generic parameters to evaluate equality of types

### PR DESCRIPTION
**Pull request by akihikodaki**
### Description of Change ###

tl; dr `Foo<bool>` and `Foo<string>` are different types

The change may be a bit confusing. `InheritsFromOrImplements` compares the results of `ResolvedCached`, but the method strips the generic arguments. . We have already compared the original types at the very beginning of `InheritsFromOrImplements` so we don't have to compare again at all.

### Issues Resolved ### 
I was too lazy to open an issue again.

### API Changes ###

 
 None

### Platforms Affected ### 

- Core/XAML (all platforms)

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### PR Checklist ###

- [X] Has automated tests 
- [ ] Rebased on top of the target branch at time of PR
- [X] Changes adhere to coding standard

### CODE REVIEW

**Comment by StephaneDelcroix** [comment]
build

**Comment by akihikodaki** [comment]
A XAML sample triggering this issue is available at https://github.com/akihikodaki/Akihiko.Xamarin.Forms.Xaml.Issues.

**Comment by akihikodaki** [comment]
fa14cd8469b492a6a728ca8fe05093950ecb0e4e resolves https://github.com/xamarin/Xamarin.Forms/pull/4062#pullrequestreview-163875361.

**Comment by StephaneDelcroix** [comment]
@akihikodaki could you rebase to fix the conflict ? thanks

**Comment by akihikodaki** [comment]
5768f1814587593481b9d4e37a13c614958c2e32: rebased to 2926b364c1bf1ebee46d683a54ab3246b68f4cda.
